Core/src/main/java/com/plotsquared/core/util/EventDispatcher.java

```diff
@@ -130,8 +133,23 @@ public PlayerAutoPlotEvent callAuto(
         return event;
     }
 
-    public PlayerTeleportToPlotEvent callTeleport(PlotPlayer<?> player, Location from, Plot plot) {
-        PlayerTeleportToPlotEvent event = new PlayerTeleportToPlotEvent(player, from, plot);
+    public PlayerAutoPlotsChosenEvent callAutoPlotsChosen(
+            PlotPlayer<?> player, List<Plot> plots
+    ) {
+        PlayerAutoPlotsChosenEvent event =
+                new PlayerAutoPlotsChosenEvent(player, plots);
+        callEvent(event);
+        return event;
+    }
+
+    public PlotClaimedNotifyEvent callPlotClaimedNotify(Plot plot, boolean auto) {
+        PlotClaimedNotifyEvent event = new PlotClaimedNotifyEvent(plot, auto);
+        callEvent(event);
+        return event;
+    }
+
+    public PlayerTeleportToPlotEvent callTeleport(PlotPlayer<?> player, Location from, Plot plot, TeleportCause cause) {
+        PlayerTeleportToPlotEvent event = new PlayerTeleportToPlotEvent(player, from, plot, cause);
```

Comment:
Javadocs

Reply:
I would rather not add any javadocs to this class... This is about as far away from API as we'd want to get and it's pretty obvious what each does to maintainers.

Reply:
Indeed, I solely like having javadocs on public methods, but we toss an `@ApiStatus.Internal` on the class to make it more obvious.